lib/sensors/time_of_flight.cpp

```diff
@@ -0,0 +1,180 @@
+    i2c_->writeByteToRegister(device_address_, 0x00e3, 0x01);
+    i2c_->writeByteToRegister(device_address_, 0x00e4, 0x03);
+    i2c_->writeByteToRegister(device_address_, 0x00e5, 0x02);
+    i2c_->writeByteToRegister(device_address_, 0x00e6, 0x01);
+    i2c_->writeByteToRegister(device_address_, 0x00e7, 0x03);
+    i2c_->writeByteToRegister(device_address_, 0x00f5, 0x02);
+    i2c_->writeByteToRegister(device_address_, 0x00d9, 0x05);
+    i2c_->writeByteToRegister(device_address_, 0x00db, 0xce);
+    i2c_->writeByteToRegister(device_address_, 0x00dc, 0x03);
+    i2c_->writeByteToRegister(device_address_, 0x00dd, 0xf8);
+    i2c_->writeByteToRegister(device_address_, 0x009f, 0x00);
+    i2c_->writeByteToRegister(device_address_, 0x00a3, 0x3c);
+    i2c_->writeByteToRegister(device_address_, 0x00b7, 0x00);
+    i2c_->writeByteToRegister(device_address_, 0x00bb, 0x3c);
+    i2c_->writeByteToRegister(device_address_, 0x00b2, 0x09);
+    i2c_->writeByteToRegister(device_address_, 0x00ca, 0x09);
+    // i2c_->writeByteToRegister(device_address_, 0x0198, 0x01);
+    // i2c_->writeByteToRegister(device_address_, 0x01b0, 0x17);
+    // i2c_->writeByteToRegister(device_address_, 0x01ad, 0x00);
+    i2c_->writeByteToRegister(device_address_, 0x00ff, 0x05);
+    // i2c_->writeByteToRegister(device_address_, 0x0100, 0x05);
+    // i2c_->writeByteToRegister(device_address_, 0x0199, 0x05);
+    // i2c_->writeByteToRegister(device_address_, 0x01a6, 0x1b);
+    // i2c_->writeByteToRegister(device_address_, 0x01ac, 0x3e);
+    // i2c_->writeByteToRegister(device_address_, 0x01a7, 0x1f);
+    i2c_->writeByteToRegister(device_address_, 0x0030, 0x00);
+
+    i2c_->writeByteToRegister(device_address_, kSystemFreshOutOfReset, 0);
+
+    // Recommended : Public registers
+    i2c_->writeByteToRegister(device_address_, kSystemModeGpioOne, 0x10);
+    // i2c_->writeByteToRegister(device_address_, kReadoutSamplingPeriod, 0x30);
+    i2c_->writeByteToRegister(device_address_, kSysAlsAnalogueGain, 0x46);
+    i2c_->writeByteToRegister(device_address_, kSysRangeVhvRepeatRate, 0xff);
+    i2c_->writeByteToRegister(device_address_, kSysAlsIntegrationPeriod, 0x63);
+    i2c_->writeByteToRegister(device_address_, kSysRangeVhvRecalibrate, 0x01);
+
+    // Optional: Public registers
+    i2c_->writeByteToRegister(device_address_, kSysRangeIntermeasurementPeriod, 0x09);
+    i2c_->writeByteToRegister(device_address_, kSysAlsIntermeasurementPeriod, 0x31);
+    i2c_->writeByteToRegister(device_address_, kSystemInterruptConfigGpio, 0x24);
+  }
+  return core::Result::kSuccess;
+}
+
+std::optional<std::uint8_t> TimeOfFlight::getRangeSingleShot()
+{
+  if (checkSensorMode(kModeSingleShot) == core::Result::kFailure) { return std::nullopt; }
+  return getRange();
+};
+
+std::optional<std::uint8_t> TimeOfFlight::getRangeContinuous()
+{
+  if (checkSensorMode(kModeContinuous) == core::Result::kFailure) { return std::nullopt; }
+  return getRange();
+};
```

Comment:
what are these for?

Reply:
The sensor has two modes of operation: Single shot, and continuous. Single shot is a single measurement, while continuous will repeatedly measure the range with a time interval based on the value of `kSysRangeIntermeasurementPeriod`. Before reading in a given mode, register `kSysRangeStart` has to be set to either 1 or 3
Because the code to check that `kSysRangeStart` has the correct value is the same whether its checking for continuous or single-shot, and because the register to read the range from is the same in both cases if `kSysRangeStart` has the correct value (`kResultRangeVal`), `getRangeSingleShot()` and `getRangeContinuous()` call these separate functions to retrieve the range

As for which value will be wanted, single shot or continous, that's yet to be decided